client/src/components/FullProduct/FullProduct.js

```diff
@@ -1,38 +1,73 @@
-import React, { Component } from 'react';
+import React from 'react';
+import { useQuery, gql } from '@apollo/client';
+
 import MiniGallery from './MiniGallery';
 import ValueAttributes from './ValueAttributes';
 
-export class FullProduct extends Component {
-  _handleKeyUp = (event) => {
-    if (event.key === 'Escape') {
-      return this.props.onShowProduct(this.props.item);
+const GET_PRODUCT_BY_ID = gql`
+  query myQuery($id: String!) {
+    product(id: $id) {
+      id
+      name
+      inStock
+      gallery
+      description
+      category
+      attributes {
+        name
+        id
+        type
+        items {
+          id
+          value
+          displayValue
+        }
+      }
+      prices {
+        amount
+        currency {
+          label
+          symbol
+        }
+      }
+      brand
     }
-  };
-  componentDidMount() {
-    document.addEventListener('keyup', this._handleKeyUp);
-  }
-  componentWillUnmount() {
-    document.removeEventListener('keyup', this._handleKeyUp);
   }
-  render() {
-    return (
+`;
+
+const FullProduct = (props) => {
+  const {
+    error: errorProduct,
+    data: dataProduct,
+    loading: loadingProduct,
+  } = useQuery(GET_PRODUCT_BY_ID, {
+    variables: {id: props.productId}
+  });
+  console.log(errorProduct)
+  return (
+    <> 
+    {loadingProduct ? <div> Loading...</div>
```

Comment:
maybe we should also add error handling?

Reply:
Sounds great! Thanks!
Done ✅👌